Islandora/islandora_defaults
# D9

**GitHub Issue**: (https://github.com/Islandora/documentation/issues/1681)



# What does this Pull Request do?

Updates for D9

# What's new?
* Updates info.ymls for D9 
* Updates permissions_by_term to a version that supports d9 (NOTE - this version ONLY supports D9 and does NOT support d8)
* Replaces getCurrentUserId with getDefaultEntityOwner due to https://www.drupal.org/project/drupal/issues/3153455
* Updates travis to php 7.4
* Updates travis to use recommended_project_d9 branch of islandora_ci (so it could build a d9 environment...)
* Temporarily points to branches of islandora/islandora and islandora/openseadragon that are for d9 support
* makes the default theme bartik not carapace
* Replaces keys with ids in migration source configuration (see https://www.drupal.org/node/3060246)


# How should this be tested?
make sure islandora_defaults still works?
* updatedb will need to be run
* due to the dependency chain - this can only be tested on a D9 site. so you could use https://github.com/Islandora-Devops/islandora-playbook/pull/196 to build a d9 site if you'd like


# Interested parties
@Islandora/8-x-committers
